feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -0,0 +1,567 @@
+			reqBody.PduSessionId = sbi.PduSessionId(m5gCtx.GetPduSessionId())
+			reqBody.Gpsi = getSbiGpsi(m5gCtx.GetGpsi())
+			reqBody.PduSessionType = getSbiPduSesionType(m5gCtx.GetPduSessionType())
+		}
+	}
+
+	return reqBody
+}
+
+func GenNotifyUrl(apiRoot string, sessionId string) sbi.Uri {
+	return sbi.Uri(fmt.Sprintf("%s/%s", apiRoot, b64.URLEncoding.EncodeToString([]byte(sessionId))))
+}
+
+func GetCreateSessionResponseProto(
+	request *protos.CreateSessionRequest,
+	smPolicyDecision *n7_sbi.SmPolicyDecision,
+	smPolicyUrl string,
+) *protos.CreateSessionResponse {
+	staticRules, dynamicRules, _ := GetPccRules(smPolicyDecision)
+	eventTriggers, revalidationTime := GetEventTriggersInfoProto(
+		smPolicyDecision.PolicyCtrlReqTriggers, smPolicyDecision.RevalidationTime)
+
+	tgppCtx := &protos.TgppContext{GxDestHost: smPolicyUrl}
+	monitors := GetUsageMonitorsProto(request, tgppCtx, eventTriggers, revalidationTime, smPolicyDecision)
+
+	rpcResp := &protos.CreateSessionResponse{
+		StaticRules:      staticRules,
+		DynamicRules:     dynamicRules,
+		TgppCtx:          tgppCtx,
+		SessionId:        request.SessionId,
+		EventTriggers:    eventTriggers,
+		RevalidationTime: revalidationTime,
+		UsageMonitors:    monitors,
+	}
+	if smPolicyDecision.Online != nil {
+		rpcResp.Online = *smPolicyDecision.Online
+	}
+	if smPolicyDecision.Offline != nil {
+		rpcResp.Offline = *smPolicyDecision.Offline
+	}
+	return rpcResp
+}
+
+func getSbiIpv4(ipv4 string) *sbi.Ipv4Addr {
+	if len(ipv4) == 0 {
+		return nil
+	}
+	ret := sbi.Ipv4Addr(ipv4)
+	return &ret
+}
+
+func getSbiIpv6(ipv6 string) *sbi.Ipv6Prefix {
+	if len(ipv6) == 0 {
+		return nil
+	}
+	ret := sbi.Ipv6Prefix(ipv6)
+	return &ret
+}
+
+func getSbiGpsi(gpsi string) *sbi.Gpsi {
+	if len(gpsi) == 0 {
+		return nil
+	}
+	ret := sbi.Gpsi(gpsi)
+	return &ret
+}
+
+func getSbiRatType(ratType protos.RATType) *sbi.RatType {
+	var sbiRatType sbi.RatType
+	switch ratType {
+	case protos.RATType_TGPP_LTE:
+		sbiRatType = sbi.RatTypeEUTRA
+	case protos.RATType_TGPP_NR:
+		sbiRatType = sbi.RatTypeNR
+	case protos.RATType_TGPP_WLAN:
+		sbiRatType = sbi.RatTypeWLAN
+	default:
+		return nil
+	}
+	return &sbiRatType
+}
+
+func getSbiAccessType(ratType protos.RATType) *sbi.AccessType {
+	var accessType sbi.AccessType
+	switch ratType {
+	case protos.RATType_TGPP_LTE:
+		accessType = sbi.AccessTypeN3GPPACCESS
+	case protos.RATType_TGPP_NR:
+		accessType = sbi.AccessTypeN3GPPACCESS
+	case protos.RATType_TGPP_WLAN:
+		accessType = sbi.AccessTypeNON3GPPACCESS
+	default:
+		return nil
+	}
+	return &accessType
+}
+
+func getSbiTimeZone(tzProto *protos.Timezone) *sbi.TimeZone {
+	if tzProto == nil {
+		return nil
+	}
+	offsetMin := tzProto.GetOffsetMinutes()
+	isNegative := false
+	if offsetMin < 0 {
+		isNegative = true
+		offsetMin = -offsetMin
+	}
+	tz := fmt.Sprintf("%02d:%02d", offsetMin/60, offsetMin%60)
+	if isNegative {
+		tz = "-" + tz
+	} else {
+		tz = "+" + tz
+	}
+	tzStr := sbi.TimeZone(tz)
+	return &tzStr
+}
```

Comment:
this function may deserve a test case by itself to make sure positive and negative and 30 min time zones are covered

Reply:
Added a test for this